osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetFirmwareVersionResponseMessageProcessor.java

```
@@ -0,0 +1,52 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.alliander.osgp.adapter.domain.smartmetering.infra.jms.core.messageprocessors;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.alliander.osgp.adapter.domain.smartmetering.application.services.ConfigurationService;
+import com.alliander.osgp.adapter.domain.smartmetering.infra.jms.core.OsgpCoreResponseMessageProcessor;
+import com.alliander.osgp.domain.core.valueobjects.DeviceFunction;
+import com.alliander.osgp.dto.valueobjects.FirmwareVersionDto;
+import com.alliander.osgp.shared.exceptionhandling.OsgpException;
+import com.alliander.osgp.shared.infra.jms.DeviceMessageMetadata;
+import com.alliander.osgp.shared.infra.jms.ResponseMessage;
+
+@Component
+public class GetFirmwareVersionResponseMessageProcessor extends OsgpCoreResponseMessageProcessor {
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    protected GetFirmwareVersionResponseMessageProcessor() {
+        super(DeviceFunction.GET_FIRMWARE_VERSION);
+    }
+
+    @Override
+    protected boolean hasRegularResponseObject(final ResponseMessage responseMessage) {
+        final Object dataObject = responseMessage.getDataObject();
+        return dataObject instanceof ArrayList;
+    }
+
+    @Override
+    protected void handleMessage(final DeviceMessageMetadata deviceMessageMetadata,
+            final ResponseMessage responseMessage, final OsgpException osgpException) {
+
+        if (responseMessage.getDataObject() instanceof ArrayList) {
+            final List<FirmwareVersionDto> firmwareVersionList = (List<FirmwareVersionDto>) responseMessage
+                    .getDataObject();
+
+            this.configurationService.handleGetFirmwareVersionResponse(deviceMessageMetadata,
+                    responseMessage.getResult(), osgpException, firmwareVersionList);
+        }
```

Maybe throw an exception when it is not an ArrayList, to not fail silently when hasRegularResponseObject was not called first.